lib/src/benchmark_base.dart

```diff
@@ -35,16 +35,21 @@ class BenchmarkBase {
   // time minimum has been reached.
   static double measureFor(Function f, int minimumMillis) {
     int minimumMicros = minimumMillis * 1000;
-    int iter = 0;
+    int iter = 1;
+    int elapsed = 0;
     Stopwatch watch = Stopwatch();
     watch.start();
-    int elapsed = 0;
-    while (elapsed < minimumMicros) {
-      f();
+    while (true) { // ignore: literal_only_boolean_expressions
+      watch.reset();
+      for (int i = 0; i < iter; i++) {
+        f();
```

Comment:
General caveate for this kind of benchmarks: If a good compiler inlines everything, it might see that `f` has no visible side-effects and eliminate it entirely. That's why the measured function should always return a result that can be checked for correctness. Merely "doing something" can be eliminated if it isn't checked.
(Nothing new about that, though).